esp/services/esdl_svc_engine/esdl_svc_custom.cpp

```diff
@@ -78,6 +78,18 @@ void CEsdlCustomTransformChoose::processClauses(IPropertyTree *request, IXpathCo
                             request->appendProp(targetField, evaluatedValue.str());
                         }
                     }
+                    catch (IException* e)
+                    {
+                        StringBuffer eMsg;
+                        e->errorMessage(eMsg);
+                        e->Release();
+
+                        VStringBuffer msg("Could not process Custom Transform: '%s' [%s]", cur.queryName(), eMsg.str());
+                        if (!optional)
+                            throw MakeStringException(-1, "%s", msg.str());
```

Comment:
Really prefer to declare new codes rather than use -1.

Reply:
I generally agree, although -1 is a frequently used code. In this case, the expectation is that the exception thrown here will be consumed by the caller. The callers currently ignore the code, using only the message text for debug logging. Even when the caller propagates an exception, it is not this exception with this code.